webpack/scenes/Subscriptions/Manifest/SimpleContentAccess.js

```diff
@@ -0,0 +1,103 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Switch } from '@patternfly/react-core';
+import { OverlayTrigger, Tooltip, Icon } from 'patternfly-react';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { Col, ControlLabel } from 'react-bootstrap';
+
+const SimpleContentAccess = (props) => {
+  const {
+    canToggleSimpleContentAccess,
+    simpleContentAccessValue,
+    enableSimpleContentAccess,
+    disableSimpleContentAccess,
+    setActionInProgress,
+    hideModal,
+  } = props;
+
+  const [isChecked, setChecked] = useState(simpleContentAccessValue);
+
+  useEffect(() => {
+    if (isChecked !== simpleContentAccessValue) {
+      setChecked(simpleContentAccessValue);
+    }
+  }, [simpleContentAccessValue]);
```

Comment:
This still feels off to me, because we're trying to keep props in sync with internal state.

Here's how `simpleContentAccessValue` currently affects the state of our switch:
1. Get the value via prop.
2. Set the state to the Prop value.
3. If the prop changes, update the state.
4. If the state changes (due to user clicking), call `disableSimpleContentAccess` or `enableSimpleContentAccess`, not based on the state, but based on the current prop value.  This gives the advantage of immediate change in response to user action, but also gives us a period of time (while the task is finishing) where the state is out of sync with props and the visual state of the switch is not accurate.

After some more thinking, want to get your thoughts on this possible alternative:

1. The Switch will not toggle immediately based on user clicks.  Instead, when the user clicks, the switch will disable without changing value and we will show a small loading spinner.
2. The loading spinner will remain (with the disabled switch) for as long as the task takes to complete.
3. When the task is complete, the loading spinner will clear and the switch state will change to reflect the completion of the task.

(The modal could still be closed for 2 and 3 if we decide to keep it that way)

This way, the state of the switch always reflects our best knowledge, and there is never a moment when anything is inaccurate.  In addition, we still get the immediate UI response upon user click.

Reply:
This approach will also simplify this component, since you won't need the `useEffect` to track the prop anymore.

Reply:
update: It looks like much of this is already implemented (the loading spinner is there).  The missing piece is that the switch shouldn't change value until the task is complete